crossfit/backend/torch/hf/memory_curve_utils.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2024 NVIDIA CORPORATION
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import gc
+
+import joblib
+import numpy as np
+import torch
+from sklearn.linear_model import LinearRegression
+from tqdm import tqdm
+from transformers import AutoConfig, AutoTokenizer, PreTrainedModel
+
+from crossfit.utils.model_adapter import adapt_model_input
+
+
+def fit_memory_estimate_curve(
+    model: PreTrainedModel,
+    path_or_name: str,
+    start_batch_size: int = 1,
+    end_batch_size: int = 2048,
+    batch_size_increment: int = 256,
+    start_seq_len: int = 1,
+    end_seq_len: int = 2048,
+    seq_len_increment: int = 64,
+    mem_model_path: str = None,
+) -> LinearRegression:
+    print(f"Fitting memory estimate curve for model: {path_or_name}")
+
+    device = next(model.parameters()).device
+    X: list[list[int]] = []
+    y: list[float] = []
+
+    max_seq = min(AutoTokenizer.from_pretrained(path_or_name).model_max_length, end_seq_len)
```

Comment:
Similar to the other issue, I think every time `AutoTokenizer` or `AutoConfig` is used in this file it should be using the corresponding methods of the model. Now I'm getting this error with llama guard:
```
Traceback (most recent call last):
  File "/usr/local/bin/aegis_classifier_inference", line 8, in <module>
    sys.exit(console_script())
  File "/usr/local/lib/python3.10/dist-packages/nemo_curator/scripts/aegis_classifier_inference.py", line 126, in console_script
    main()
  File "/usr/local/lib/python3.10/dist-packages/nemo_curator/scripts/aegis_classifier_inference.py", line 62, in main
    domain_classifier = AegisClassifier(
  File "/usr/local/lib/python3.10/dist-packages/nemo_curator/classifiers/aegis.py", line 161, in __init__
    model = AegisHFModel(config=config)
  File "/usr/local/lib/python3.10/dist-packages/nemo_curator/classifiers/aegis.py", line 90, in __init__
    super().__init__(
  File "/usr/local/lib/python3.10/dist-packages/crossfit/backend/torch/hf/model.py", line 58, in __init__
    self.mem = fit_memory_estimate_curve(
  File "/usr/local/lib/python3.10/dist-packages/crossfit/backend/torch/hf/memory_curve_utils.py", line 44, in fit_memory_estimate_curve
    max_seq = min(AutoTokenizer.from_pretrained(path_or_name).model_max_length, end_seq_len)
  File "/usr/local/lib/python3.10/dist-packages/transformers/models/auto/tokenization_auto.py", line 843, in from_pretrained
    return tokenizer_class_fast.from_pretrained(pretrained_model_name_or_path, *inputs, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/transformers/tokenization_utils_base.py", line 2032, in from_pretrained
    raise EnvironmentError(
OSError: Can't load tokenizer for 'meta-llama/LlamaGuard-7b'. If you were trying to load it from 'https://huggingface.co/models', make sure you don't have a local directory with the same name. Otherwise, make sure 'meta-llama/LlamaGuard-7b' is the correct path to a directory containing all relevant files for a LlamaTokenizerFast tokenizer.
```

Reply:
Resolved by: https://github.com/rapidsai/crossfit/pull/66/commits/987836f8aa52b596bd053f33af240573d50fe267